src/System.Drawing.Common/src/System/Drawing/Brush.cs

```diff
@@ -100,34 +60,13 @@ protected virtual void Dispose(bool disposing)
                     Debug.Assert(status == SafeNativeMethods.Gdip.Ok, "GDI+ returned an error status: " + status.ToString(CultureInfo.InvariantCulture));
 #endif
                 }
-                catch (Exception ex)
-                {
-                    if (ClientUtils.IsSecurityOrCriticalException(ex))
-                    {
-                        throw;
-                    }
-
-                    Debug.Fail("Exception thrown during Dispose: " + ex.ToString());
-                }
```

Comment:
thats a fair point about entrypointnotfound. We could get this in Windows Nano

I thought that the debug.fail indicates that we really really don't expect any exception, but we've seen some fault assertions in the library already as it seems no one ever tested it in debug mode.

Do you reckon I should revert this? I didn't quite understand your comment about mono.